Made module exclusively Python3

AidanDelaney opened the pull request:
Hi,
First of all, thank you for this module.  It has saved us a good chunk of development time.  I've updated this module so  that it is Python 3 only.  I've also added another test and updated it for Flask > 1.0.  Hooks can now return their http status, which the runner is free to ignore, or return to the end-user.  Please let me know if you'd like me to change this PR in any way.

Thank you.

AidanDelaney commented:
This branch should now cleanly merge onto master.  It removes Python 2 runtimes from tox, and Travis.

alexchamberlain commented on github_webhook/__init__.py:
```diff
@@ -8,6 +8,15 @@
+
+import sys
+if sys.version_info[0] < 3:
+    raise Exception(
```
Consider RuntimeError.

AidanDelaney replied:
Fixed

alexchamberlain commented on setup.py:
```diff
@@ -1,14 +1,14 @@
       license='Apache 2.0',
       packages=["github_webhook"],
-      install_requires=['flask', 'six'],
+      install_requires=['flask==1.0.2'],
```
Consider `>= 1.0.2`.

AidanDelaney replied:
Fixed.

alexchamberlain commented on github_webhook/webhook.py:
```diff
@@ -69,7 +69,9 @@ def _postreceive(self):
 
         for hook in self._hooks.get(event_type, []):
-            hook(data)
+            resp = hook(data)
```
If the use case is sending an error, would this be better represented with an exception?

AidanDelaney replied:
In this case we're returning the http code that the wrapped function returns.  This could be any http code such as a 301 which is not necessarily an error.  I think the current approach is the best choice here, but I'm happy to be corrected.

alexchamberlain commented on setup.py:
```diff
@@ -9,7 +9,7 @@
       packages=["github_webhook"],
       install_requires=['flask==1.0.2'],
-      tests_require=['mock', 'nose'],
+      tests_require=['mock', 'pytest', 'nose'],
```
We should remove `mock` and `nose` here.

AidanDelaney replied:
fixed

alexchamberlain commented on tox.ini:
```diff
@@ -1,13 +1,12 @@
      flask
-     six
-     py{27,py}: mock
+     pypy: mock
```
If we're Python 3 only, is `pypy` appropriate?

AidanDelaney replied:
I've changed it to pypy3, but is there any real reason to support pypy?

alexchamberlain commented on tox.ini:
```diff
@@ -1,12 +1,12 @@
      pytest-cov
      flask
-     pypy: mock
+     mock
```
We can remove `mock`.

AidanDelaney replied:
Fixed

alexchamberlain reviewed (commented):
Overall, this LGTM - a few comments on the boundaries.

AidanDelaney committed:
Run with updated Flask API and Python 3 only

This change uses the post 1.0 version of the Flask API.  We drop
support for Python 2 which allows us to remove dependencies on
`mock` and `six`.  Furthermore, we test against the documented
`postevent` from GitHub.  We choose to use the entire object
rather than a `postevent`-like object in order to test against
"real world" data.

Signed-off-by: Aidan Delaney <adelaney21@bloomberg.net>

AidanDelaney committed:
Run test environment without legacy dependencies

Changed the error thrown on detecting a legacy Python 2 environment
to be a RuntimeError.  Also removed tox dependency on mock and made
the Flask dependency less strict.  We now depend on any Flask release
greater than 1.0.x.

Signed-off-by: Aidan Delaney <adelaney21@bloomberg.net>
